assert/assertions.go

```diff
@@ -109,7 +109,19 @@ func copyExportedFields(expected interface{}) interface{} {
 		result.Elem().Set(reflect.ValueOf(unexportedRemoved))
 		return result.Interface()
 
-	case reflect.Array, reflect.Slice:
+	case reflect.Array:
+		result := reflect.New(reflect.ArrayOf(expectedValue.Len(), expectedType.Elem())).Elem()
+		for i := 0; i < expectedValue.Len(); i++ {
+			index := expectedValue.Index(i)
+			if isNil(index) {
+				continue
```

Comment:
This case is not covered by the test suite.

Reply:
Thanks!

I added one more test for nil-value, but it failed. I fixed it by passing an actual `interface{}` to the isNil function.